src/include/rados/librados.hpp

```diff
@@ -1040,6 +1040,7 @@ namespace librados
     friend class ObjectWriteOperation;  // copy_from needs to see our IoCtxImpl
 
     IoCtxImpl *io_ctx_impl;
+    std::list<std::string> *cached_pool_names;
```

Comment:
On Thu, 5 Feb 2015, rzarzynski wrote:
> 
> In src/include/rados/librados.hpp:
> 
> > @@ -1040,6 +1040,7 @@ namespace librados
> >      friend class ObjectWriteOperation;  // copy_from needs to see our IoCtxImpl
> >  
> >      IoCtxImpl *io_ctx_impl;
> > +    std::list<std::string> *cached_pool_names;
> 
> Poking with raw pointers is error prone, I know. Storing the std::list just as the member of IoCtx would be much better
> solution. Unfortunately, it looks like the const variant of get_pool_name() must be able to modify the list too. Maybe
> could we address that via some kind of smart pointer? On the other hand the librados doesn't seem to use such things
> currently, so it could be an introduction of unwanted dependency...

Ah, that's what the 'mutable' keyword is for:

 mutable std::list<std::string> cached_pool_names;